src/domain/BattleField.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @typedef {import('./units/Player').Player} Player
+ * @typedef {import('./core/units/Monster').Monster} Monster
+ * @typedef {import('./core/skills/Skill').Skill} Skill
+ */
+
+import { ERROR_MESSAGE } from '../constants/error.js';
+import { Monster } from './core/units/Monster.js';
+import { Player } from './units/Player.js';
+
+export class BattleFiled {
+  #turn = 1;
+
+  #player;
+
+  #enemy;
+
+  #winner = null;
+
+  #disable = false;
+
+  /**
+   * @param {Player} player
+   */
+  constructor(player) {
+    this.#validate(player);
+    this.#player = player;
+  }
+
+  #validate(player) {
+    if (!(player instanceof Player)) {
+      throw new Error(ERROR_MESSAGE.INVALID_PLAYER);
+    }
+  }
+
+  /**
+   * @returns {number}
+   */
+  get turn() {
+    return this.#turn;
+  }
+
+  /**
+   * @returns {Player}
+   */
+  get player() {
+    return this.#player;
+  }
+
+  /**
+   * @returns {Monster}
+   */
+  get enemy() {
+    return this.#enemy;
+  }
+
+  /**
+   * @returns {Monster | Player}
+   */
+  get winner() {
+    return this.#winner;
+  }
```

Comment:
```suggestion
checkRaidEnd() {
    return this.#winner == null;
  }
```

상위 클래스에서 winner을 get해 레이드가 끝났는지 확인하기 보다는 클래스 내부에서 레이드 종료를 확인하는건 어떨까요?

Reply:
지금 생각해보니 해당 방법이 확실히 객체에 메세지를 전달하는 듯한 느낌이네요!